app/controllers/foreman_tasks/recurring_logics_controller.rb

```diff
@@ -22,6 +22,16 @@ def cancel
       redirect_to :action => :index
     end
 
+    def clear_cancelled
+      scope = resource_base.search_for('state=cancelled') || []
+      scope.each do |recurring_logic|
```

Comment:
I think you should be able to call `#destroy_all` on the collection instead of having to iterate through the items and destroying them one by one